CMakeLists.txt

```diff
@@ -80,7 +82,11 @@ mark_as_advanced(
 if(WIN32)
   #set(ROR_USING_VISUAL_STUDIO_PROFESSIONAL "FALSE" CACHE BOOL "if you use VS professional, enable this. DO NOT enable this when using express versions")
   
-  set(warnings /W3 /D_CRT_SECURE_NO_WARNINGS /wd4005 /wd4996 /wd4251 /wd4275 /wd4099 /nologo)
+  if (BUILD_SHOW_WARNINGS)
+    set(warnings /W3 /D_CRT_SECURE_NO_WARNINGS /wd4005 /wd4996 /wd4251 /wd4275 /wd4099 /nologo)

```

Comment:
You are disabling some warnings even if the BUILD_SHOW_WARNINGS option is on. Why? What are those? There should be a code comment explaining why we want to disable these.